### ISSUE TITLE
Print to PDF with transparent background

### ISSUE THREAD

**Issue by armellarcier** [puppeteer/puppeteer#2545]
This is a feature request.

I'm currently developing a small pdf generator tool and I really need custom headers and footers on multipage documents. 

So I print (page.pdf()) the document with margins but no header/footer.

Then I print the header and the footer

Third, I use an other js library (hummus) to put the header and footer on top of every page of the multipage doc.

BUT! the printed header and footer have a white background. This seems ok to not support transparent backgrounds for printing but hey, here is a use case. ;)

Thanks for the great work.

### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 1.3.0
* Platform / OS version: Mac OSX 10.12.6
* Node.js version: 9.5.0

**Comment by yonishostak** [comment]
+1, need this too

**Comment by UNDEADgtr** [comment]
+1, need this too

**Comment by genevien** [comment]
+1, me too

**Comment by anttiviljami** [comment]
Here's a working proof-of-concept for printing pdf with transparent background:

```
await page.emulateMedia('screen');
await page._emulationManager._client.send(
  'Emulation.setDefaultBackgroundColorOverride',
  { color: { r: 0, g: 0, b: 0, a: 0 } }
);
await page.pdf({ path: 'transparent.pdf', printBackground: true });
```

Have a go :)

**Comment by UNDEADgtr** [comment]
Thanks, @anttiviljami ! It works for me!

**Comment by armellarcier** [comment]
Thanks a lot @anttiviljami ! Our pdf webservice is up and running thx to U!

**Comment by abdelilah** [comment]
There's a slight problem with that solution when enabling displayHeaderFooter:

```
await page.emulateMedia('screen');
await page._emulationManager._client.send(
  'Emulation.setDefaultBackgroundColorOverride',
  { color: { r: 0, g: 0, b: 0, a: 0 } }
);
await page.pdf({
    path: 'transparent.pdf',
    printBackground: true,
    displayHeaderFooter: true // White background comes back after setting this
});
```

**Comment by kblok** [comment]
@aslushnikov any plans to implement this with a PDF option?

**Comment by aslushnikov** [comment]
> any plans to implement this with a PDF option?

@kblok no, we don't have any plans - folks seem to be happy with the workaround.

**Comment by septatrix** [comment]
The workaround seems to work though it took me some time to find it. Would it be possible to state this in the documentation (next to the background option) or add as a separate option (which probably could just to the above)?

**Comment by septatrix** [comment]
As the workaround **does not work** for neither **printing with a header/footer** included nor when trying to print elements which have a **background but removing them with `printBackground: false`** (they become boxes with a white background while the rest is correctly transparent) here are the corresponding bugs in the chromium bug tracker so you can star them there so they get prioritized:


[Printed pdf is not transparent even though background is](https://bugs.chromium.org/p/chromium/issues/detail?id=1078851)
[Printing inserts a white page background colour](https://bugs.chromium.org/p/chromium/issues/detail?id=498892)

**Comment by gavin310** [comment]
For what it's worth, I think supporting transparent backgrounds should definitely be a feature without having to use hacks. I understand it's still just a feature request in Chromium, but if Puppeteer could wrap the workaround into an option until Chromium supports it, that would be ideal.

**Comment by d-buckner** [comment]
I've added this functionality in the PR above. It's been included in [v9.0.0](https://pptr.dev/#?product=Puppeteer&version=v9.0.0&show=api-pagepdfoptions) with the `omitBackground` option.

**Comment by Errors4l** [comment]
@d-rowe Since this workaround/solution still doesn't work when combined with `displayHeaderFooter`, would it be a good idea to add a note on the [api.md documentation page](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md)?

For other readers finding this after struggling with the same issue, please read [septatrix's original comment](https://github.com/puppeteer/puppeteer/issues/2545#issuecomment-645660532). Star the Chromium issue and hopefully they'll fix the issue some day (it's been open since 2015, sigh).
